server/src/main/java/org/elasticsearch/common/joda/JodaDateFormatter.java

```diff
@@ -33,16 +33,21 @@
 import java.util.Locale;
 
 public class JodaDateFormatter implements DateFormatter {
-    final String pattern;
 
+    final String pattern;
+    private final int year;
     final DateTimeFormatter parser;
-
     final DateTimeFormatter printer;
 
-    public JodaDateFormatter(String pattern, DateTimeFormatter parser, DateTimeFormatter printer) {
+    JodaDateFormatter(String pattern, DateTimeFormatter parser, DateTimeFormatter printer) {
+        this(pattern, parser, printer, 1970);
+    }
+
+    private JodaDateFormatter(String pattern, DateTimeFormatter parser, DateTimeFormatter printer, int year) {
         this.pattern = pattern;
-        this.printer = printer.withDefaultYear(1970);
-        this.parser = parser.withDefaultYear(1970);
+        this.year = year;
```

Comment:
instead of passing this in, can the ctor callers be changed to callwithDefaultYear before passing in the parser and printer? there are only a places, all in Joda.java?

Reply:
good one. done.